fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/identifiers/HashConverter.java

```diff
@@ -26,22 +26,32 @@
 public class HashConverter extends InternalIdentifierConverter {
 
     @Override
-    protected String doForward(final String externalId) {
+    public String apply(final String externalId) {
+        if (!inDomain(externalId)) {
```

Comment:
This is a bit confusing. I get the logic, but a comment would be good here, about how in(or out)-of-domain-ness, in this particular case, is actually about the existence of a `#`.